met-api/src/met_api/models/comment.py

```diff
@@ -0,0 +1,75 @@
+"""Comment model class.
+
+Manages the comment
+"""
+from datetime import datetime
+from sqlalchemy import and_
+from sqlalchemy.sql.schema import ForeignKey
+from met_api.constants.comment_status import Status
+from met_api.models.engagement import Engagement
+from met_api.models.survey import Survey
+from .comment_status import CommentStatus
+from .db import db
+from .default_method_result import DefaultMethodResult
+
+
+class Comment(db.Model):
+    """Definition of the Comment entity."""
+
+    __tablename__ = 'comment'
+    id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    text = db.Column(db.Text, unique=False, nullable=False)
+    submission_date = db.Column(db.DateTime)
+    reviewed_by = db.Column(db.String(50))
+    review_date = db.Column(db.DateTime)
+    status_id = db.Column(db.Integer, ForeignKey('comment_status.id', ondelete='SET NULL'))
+    survey_id = db.Column(db.Integer, ForeignKey('survey.id', ondelete='CASCADE'), nullable=False)
+    user_id = db.Column(db.Integer, ForeignKey('user.id', ondelete='SET NULL'), nullable=True)
+
+    @classmethod
+    def get_comment(cls, comment_id):
+        """Get a comment."""
+        return db.session.query(Comment).join(CommentStatus).join(Survey).filter_by(id=comment_id).first()
+
+    @classmethod
+    def get_comments_by_survey_id_query(cls, survey_id):
+        """Get all comments."""
+        return db.session.query(Comment)\
+            .join(CommentStatus)\
+            .join(Survey)\
+            .filter(Comment.survey_id == survey_id)\
+            .order_by(Comment.id.asc()).all()
```

Comment:
can remove the query at the end of the method name
also do u need all these joins?
wont just filter Comment.Survey_id == survey id work?

Reply:
as replied in the above comment, I need info from the Survey and the Comment namely the Survey name and I need the comment status name for each comment. https://xd.adobe.com/view/ae75b5d1-8424-4323-be5a-8bf2fe0112f3-fb26/screen/bdd3fdbf-9dbc-48af-ae96-8c2670452555/